File: projects/Занько/yana3/app/src/main/java/com/example/lr3/ui/list/PostListFragment.java
package com.example.lr3.ui.list;

import android.app.AlertDialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.lr3.R;
import com.example.lr3.data.RetrofitClient;
import com.example.lr3.databinding.FragmentPostListBinding;
import com.example.lr3.model.Post;
import com.example.lr3.model.PostsResponse;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class PostListFragment extends Fragment {

    private FragmentPostListBinding binding;
    private PostAdapter adapter;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentPostListBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        adapter = new PostAdapter(this::openDetail);

        binding.recycler.setLayoutManager(new LinearLayoutManager(requireContext()));
        binding.recycler.setAdapter(adapter);

        binding.btnLoad.setOnClickListener(v -> loadData());

        binding.btnTask.setOnClickListener(v -> new AlertDialog.Builder(requireContext())
                .setTitle(getString(R.string.lab_title))
                .setMessage(getString(R.string.task_text))
                .setPositiveButton("OK", (d, w) -> d.dismiss())
                .show());
    }

    private void setLoading(boolean loading) {
        binding.progress.setVisibility(loading ? View.VISIBLE : View.GONE);
        binding.btnLoad.setEnabled(!loading);
    }

    private void loadData() {
        setLoading(true);

        RetrofitClient.api().getPosts().enqueue(new Callback<PostsResponse>() {
            @Override
            public void onResponse(@NonNull Call<PostsResponse> call, @NonNull Response<PostsResponse> response) {
                setLoading(false);
                if (response.isSuccessful() && response.body() != null && response.body().getPosts() != null) {
                    List<Post> posts = response.body().getPosts();
                    adapter.submit(posts);
                    Toast.makeText(requireContext(), "Загружено: " + posts.size(), Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(requireContext(), "Ошибка ответа сервера: " + response.code(), Toast.LENGTH_LONG).show();
                }
            }

            @Override
            public void onFailure(@NonNull Call<PostsResponse> call, @NonNull Throwable t) {
                setLoading(false);
                Toast.makeText(requireContext(), "Ошибка сети: " + t.getMessage(), Toast.LENGTH_LONG).show();
            }
        });
    }

    private void openDetail(Post post) {
        NavController nav = NavHostFragment.findNavController(this);
        Bundle args = new Bundle();
        args.putSerializable("post", post);
        nav.navigate(R.id.postDetailFragment, args);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: projects/Ярома/LAB3/app/src/main/java/com/example/lab3/PostListFragment.kt
package com.example.lab3

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.lab3.databinding.FragmentPostListBinding
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class PostListFragment : Fragment() {

    private var _binding: FragmentPostListBinding? = null
    private val binding get() = _binding!!

    private val postList = mutableListOf<Post>()
    private lateinit var adapter: PostAdapter

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentPostListBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        adapter = PostAdapter(postList) { post ->
            val intent = Intent(requireContext(), DetailActivity::class.java)
            intent.putExtra("post_id", post.id)
            intent.putExtra("post_userId", post.userId)
            intent.putExtra("post_title", post.title)
            intent.putExtra("post_body", post.body)
            startActivity(intent)
        }

        binding.recyclerView.layoutManager = LinearLayoutManager(requireContext())
        binding.recyclerView.adapter = adapter

        binding.btnLoad.setOnClickListener {
            loadPosts()
        }
    }

    private fun loadPosts() {
        binding.progressBar.visibility = View.VISIBLE
        binding.tvStatus.text = "Загрузка данных с сервера..."

        RetrofitClient.apiService.getPosts().enqueue(object : Callback<List<Post>> {
            override fun onResponse(call: Call<List<Post>>, response: Response<List<Post>>) {
                binding.progressBar.visibility = View.GONE

                if (response.isSuccessful && response.body() != null) {
                    postList.clear()
                    postList.addAll(response.body()!!)
                    adapter.notifyDataSetChanged()
                    binding.tvStatus.text = "Данные успешно загружены. Элементов: ${postList.size}"
                } else {
                    binding.tvStatus.text = "Ошибка ответа сервера"
                    Toast.makeText(requireContext(), "Ошибка ответа сервера", Toast.LENGTH_LONG).show()
                }
            }

            override fun onFailure(call: Call<List<Post>>, t: Throwable) {
                binding.progressBar.visibility = View.GONE
                binding.tvStatus.text = "Не удалось загрузить данные"
                Toast.makeText(requireContext(), "Ошибка загрузки: ${t.message}", Toast.LENGTH_LONG).show()
            }
        })
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}